src/components/input/StateEditor.ts

```diff
@@ -0,0 +1,51 @@
+import { State } from "./State"
+
+export class StateEditor implements State {
```

Comment:
Maybe this should actually not implement state but rather have a property returning a copy of the current state?